SwiftLeeds/Views/Local/LocalView.swift

```diff
@@ -7,29 +7,66 @@
 
 import SwiftUI
 import MapKit
+
 struct LocalView: View {
-    @State private var bottomSheetShown = true
-    @State private var mapRegion: MKCoordinateRegion = MKCoordinateRegion(center: CLLocationCoordinate2D(latitude: 53.801277, longitude: -1.548567), span: MKCoordinateSpan(latitudeDelta: 0.1, longitudeDelta: 0.1))
     @StateObject private var model = LocalViewModel()
 
-    
+    //-1.548567
+    @State private var bottomSheetShown = true
+    @State private var mapRegion: MKCoordinateRegion = MKCoordinateRegion(center: CLLocationCoordinate2D(latitude: 53.78613099154973, longitude: -1.5461652186147719), span: MKCoordinateSpan(latitudeDelta: 0.04, longitudeDelta: 0.04))
+    @State private var selectedLocation: Local.Location?
+
     var body: some View {
-        GeometryReader{ geometry in
-            ZStack {
-                Map(coordinateRegion: $mapRegion)
+        ZStack {
+            GeometryReader{ geometry in
+                if let category = model.selectedCategory {
+                    Map(coordinateRegion: $mapRegion, showsUserLocation: true, annotationItems: category.locations) { location in
```

Comment:
A neat way to get around this if let with the same view would be to simply publish `currentLocations` from the model instead, allowing one `Map` with one array, instead of having the two.  

This would mean shifting `selectedLocation` into the model too, and then wherever you set `selectedLocation` you could just call `setLocation(location)` or use a `didSet`, or I'm sure there's a bunch of other ways to do the same thing!

Reply:
Refactored